doc/redeemer.md

```diff
@@ -0,0 +1,60 @@
+
+\**A more detailed description about max float and it's relation to a broadcaster's reserve can be found in the [PM protocol spec](https://github.com/livepeer/wiki/blob/master/spec/streamflow/pm.md#reserve).*
+
+\**This document uses the term `sender`, it can be used interchangeably with Broadcaster.*
+
+## TicketQueue
+
+1. The `ticketQueue` is a loop that runs everytime a new block is seen. It will then pop tickets off the queue starting with the oldest ticket first, and sends it to the `LocalSenderMonitor` for redemption if the `recipientRand` for the ticket has expired. 
+
+2. When the `LocalSenderMonitor` receives a ticket from the `ticketQueue` it will substract `ticket.faceValue` from the outstanding `maxFloat` as long as the ticket is in limbo. 
+
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; _This will trigger a `LocalSenderMonitor.SubscribeMaxFloatChange(ticket.sender)` notification_
+
+3. The ticket is sent to a remote Ethereum node for redemption
+
+4. When the Ethereum node returns a response the `ticket.faceValue` is added to the `maxFloat` again as the ticket is no longer in limbo.
+
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; _This will trigger a `LocalSenderMonitor.SubscribeMaxFloatChange(ticket.sender)` notification_
+
+## Monitoring Max Float
+
+1. When max float for a `sender` is requested from the `RedeemerClient` but no local cache is available, an (unary) RPC call will be sent to the `Redeemer`. 
+
+2. A second RPC call to `MonitorMaxFloat(sender)` will open up a server-side gRPC stream to receive future update. 
+
+_If this call fails the response from step 1 is returned, but not kept in cache to to prevent it becoming stale due to not being able to receive further updates_
+
+3. The `Redeemer` goroutine started by the RPC call in step 2 will start a subscription to listen for max float changes from the `LocalSenderMonitor` for the specified `sender` using `LocalSenderMonitor.SubscribeMaxFloatChange(sender)`.
+
+_Each open server-side stream will have its own subscription that will be closed when the client closes the stream. This means that each client will have a subscription for each sender it is interested in._ 
+
+4. Once the subscription from step 3 emits an event that indicates a state change for the specified `sender`, the `Redeemer` will invoke `LocalSenderMonitor.MaxFloat(sender)` to fetch the latest value. 
+
+5. Upon retrieving the latest max float value for `sender` it will be sent over the server-side gRPC stream.
+
+6. Upon receiving a `MaxFloatUpdate` over the server-side gRPC stream for `sender` it will update its local cache for that `sender` accordingly. 
+
+7. Subsequent calls to `RedeemerClient.MaxFloat(sender)` will return the locally cached value for `sender` as long as it remains available. 
+
+8. The local cache for `sender` will be cleaned up if is not requested for 5 minutes. 
+
+![Ticket Flow](./assets/redeemer/ticketflow.png)
```

Comment:
I really like these sequence diagrams, but 200-300 KB is kind of large. I suppose if we're going to be doing a lot more of these types of checkins, we should look at a blob storage extension such as git-lfs (which Github supports).

Reply:
Do you want to include this in this PR? I think this is the only remaining issue for this PR, right?

Reply:
Up to you, but I'm generally mindful of repo growth especially with large binary blobs in Git. Another option: since the source for these diagrams has been checked in and can be reproduced [1], we could upload the images themselves to Github (drag+drop the image in this text box) and link within the docs?

[1]  Maybe add a comment to the source(s) noting which tools / service can interpret the code for these diagrams?

Reply:
now using git lfs for `.png` files